provider/cldr/src/transform/calendar/japanese.rs

```diff
@@ -0,0 +1,295 @@
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::cldr_serde;
+use crate::error::Error;
+use crate::reader::open_reader;
+use crate::CldrPaths;
+use icu_locid_macros::langid;
+use icu_provider::iter::{IterableDataProviderCore, KeyedDataProvider};
+use icu_provider::prelude::*;
+use icu_provider::resource_key;
+use icu_provider::yoke::{self, *};
+use litemap::LiteMap;
+use std::convert::TryFrom;
+use std::env;
+use std::str::FromStr;
+use tinystr::TinyStr16;
+
+/// A [`ResourceKey`] to [`JapaneseErasV1`]
+// TODO (#1116) move this into icu_calendars
+pub const JAPANESE_ERAS_V1: ResourceKey = resource_key!(Calendar, "japanese", 1);
+
+const JAPANESE_FILE: &str = include_str!("./snapshot-japanese@1.json");
+/// All keys that this module is able to produce.
+pub const ALL_KEYS: [ResourceKey; 1] = [JAPANESE_ERAS_V1];
+
+/// The date at which an era started
+///
+/// The order of fields in this struct is important!
+// TODO (#1116) move this into icu_calendars
+#[derive(
+    Copy,
+    Clone,
+    PartialEq,
+    PartialOrd,
+    Eq,
+    Ord,
+    Hash,
+    Debug,
+    serde::Serialize,
+    serde::Deserialize,
+    Yokeable,
+    ZeroCopyFrom,
+)]
+pub struct EraStartDate {
+    year: i16,
+    month: u8,
+    day: u8,
+}
+
+#[icu_provider::data_struct]
+#[derive(Debug, PartialEq, Clone, Default, serde::Serialize, serde::Deserialize)]
+#[yoke(cloning_zcf)]
+// TODO (#1116) move this into icu_calendars (and make it zero-copy)
+pub struct JapaneseErasV1 {
+    pub dates_to_historical_eras: LiteMap<EraStartDate, TinyStr16>,
+    pub dates_to_eras: LiteMap<EraStartDate, TinyStr16>,
+}
+
+/// Common code for a data provider reading from CLDR JSON dates files.
+#[derive(PartialEq, Debug, Default)]
+pub struct JapaneseErasProvider {
+    // Maps of era start dates to their names
+    // We store historical_data separately for performance
+    historical_data: LiteMap<EraStartDate, TinyStr16>,
+    data: LiteMap<EraStartDate, TinyStr16>,
+}
+
+impl TryFrom<&dyn CldrPaths> for JapaneseErasProvider {
+    type Error = Error;
+    fn try_from(cldr_paths: &dyn CldrPaths) -> Result<Self, Self::Error> {
+        let mut historical_data = LiteMap::new();
+        let mut data = LiteMap::new();
+
+        let era_names_path = cldr_paths
+            .cldr_dates("japanese")?
+            .join("main")
+            .join("en")
+            .join("ca-japanese.json");
+        let era_dates_path = cldr_paths
+            .cldr_core()?
+            .join("supplemental")
+            .join("calendarData.json");
+
+        let era_names: cldr_serde::ca::Resource =
+            serde_json::from_reader(open_reader(&era_names_path)?)
+                .map_err(|e| (e, era_names_path))?;
+        let era_dates: cldr_serde::japanese::Resource =
+            serde_json::from_reader(open_reader(&era_dates_path)?)
+                .map_err(|e| (e, era_dates_path))?;
+
+        let era_name_map = &era_names
```

Comment:
`nit`: can you document why are you retrieving `en` data here (basically that we need an example, and we use `en`).